### PR TITLE
Neural networks - torch

### PR DESCRIPTION
**Pull request by kuba1302**

### CODE REVIEW

File: docker-compose.yml
Comment (jzmujdzin) [inline docker-compose.yml]: typo: ticket-upgrade-prediction-gpu

File: ticket_upgrade_prediction/evaluator.py
Comment (jzmujdzin) [inline ticket_upgrade_prediction/evaluator.py]: maybe we should move BaseEvaluator class to another file in evaluator dictory, like BaseModel class is made? what do you think about that idea?

File: ticket_upgrade_prediction/models/neural_net_torch/train.py
Comment (jzmujdzin) [inline ticket_upgrade_prediction/models/neural_net_torch/train.py]: what do you think about splitting this into another function?

File: ticket_upgrade_prediction/models/neural_net_torch/train.py
Comment (jzmujdzin) [inline ticket_upgrade_prediction/models/neural_net_torch/train.py]: what do you think about splitting this into another function?

File: ticket_upgrade_prediction/pipeline.py
Comment (jzmujdzin) [inline ticket_upgrade_prediction/pipeline.py]: `Traceback (most recent call last):
  File "E:\anaconda\lib\site-packages\IPython\core\interactiveshell.py", line 3398, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-2-7e68d98006e0>", line 1, in <cell line: 1>
    runfile('E:/wec_2022/ticket_upgrade_prediction/pipeline.py', wdir='E:/wec_2022/ticket_upgrade_prediction')
  File "E:\jetbrains\PyCharm 2022.2.1\plugins\python\helpers\pydev\_pydev_bundle\pydev_umd.py", line 198, in runfile
    pydev_imports.execfile(filename, global_vars, local_vars)  # execute the script
  File "E:\jetbrains\PyCharm 2022.2.1\plugins\python\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "E:/wec_2022/ticket_upgrade_prediction/pipeline.py", line 12, in <module>
    class Pipeline:
  File "E:/wec_2022/ticket_upgrade_prediction/pipeline.py", line 30, in Pipeline
    def read_all_files(self) -> tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame]:
TypeError: 'type' object is not subscriptable`

Comment (jzmujdzin) [reply ticket_upgrade_prediction/pipeline.py]: this line causes error

**Review by jzmujdzin — CHANGES_REQUESTED** [review]

File: ticket_upgrade_prediction/evaluator.py
Comment (kuba1302) [reply ticket_upgrade_prediction/evaluator.py]: Hmm, on first tought, i felt like it should be here, since there will be only one evaluator. Yet, if we consider AutoML for future WEC, it mind be useful for regression evaluator

File: ticket_upgrade_prediction/pipeline.py
Comment (kuba1302) [reply ticket_upgrade_prediction/pipeline.py]: which python version are You using? Older ones, doesn't allow typing like that using built in list

Comment (kuba1302) [reply ticket_upgrade_prediction/pipeline.py]: In that situation it would be best to use:
`from typing import Tuple`
It should work then. 
So i can change it, or You can use newer python version, your call

**Comment by kuba1302** [comment]
Let's first merge Your's pipeline changes, so I can adjust this PR before merging it

File: ticket_upgrade_prediction/models/neural_net_torch/neural_net_model.py
Comment (jzmujdzin) [inline ticket_upgrade_prediction/models/neural_net_torch/neural_net_model.py]: maybe log this instead of printing?

File: ticket_upgrade_prediction/models/neural_net_torch/neural_net_model.py
Comment (jzmujdzin) [inline ticket_upgrade_prediction/models/neural_net_torch/neural_net_model.py]: did you mean forward for the name of function?

File: ticket_upgrade_prediction/models/neural_net_torch/neural_net_model.py
Comment (jzmujdzin) [inline ticket_upgrade_prediction/models/neural_net_torch/neural_net_model.py]: i think this should be adjustable when initiating object of that class

File: ticket_upgrade_prediction/models/base.py
Comment (jzmujdzin) [inline ticket_upgrade_prediction/models/base.py:24]: is this used anywhere? whats the purpose

File: ticket_upgrade_prediction/models/neural_net_torch/train.py
Comment (jzmujdzin) [inline ticket_upgrade_prediction/models/neural_net_torch/train.py]: i think this could be just to_mlflow = bool(mlflow_run_name)

File: ticket_upgrade_prediction/models/neural_net_torch/train.py
Comment (jzmujdzin) [inline ticket_upgrade_prediction/models/neural_net_torch/train.py:172]: i believe it would be better if we could choose if we want stratified or non stratified k fold, then assign it to a var like k_fold and iterate over it

File: ticket_upgrade_prediction/models/neural_net_torch/train.py
Comment (jzmujdzin) [inline ticket_upgrade_prediction/models/neural_net_torch/train.py]: should mlflow_run_name be integer?

File: ticket_upgrade_prediction/models/neural_net_torch/train.py
Comment (jzmujdzin) [inline ticket_upgrade_prediction/models/neural_net_torch/train.py]: if mlflow_run_name is a boolean value (which is misleading btw) we could just do to_mlflow = mlflow_run_name

**Review by jzmujdzin — CHANGES_REQUESTED** [review]

File: ticket_upgrade_prediction/models/neural_net_torch/neural_net_model.py
Comment (kuba1302) [reply ticket_upgrade_prediction/models/neural_net_torch/neural_net_model.py]: Nothing will be better, it was for debugging purposes

File: ticket_upgrade_prediction/models/base.py
Comment (kuba1302) [reply ticket_upgrade_prediction/models/base.py:24]: Im testing it's usage for typing

File: ticket_upgrade_prediction/models/neural_net_torch/neural_net_model.py
Comment (kuba1302) [reply ticket_upgrade_prediction/models/neural_net_torch/neural_net_model.py]: good point, idk how did it work XD

File: ticket_upgrade_prediction/models/neural_net_torch/neural_net_model.py
Comment (kuba1302) [reply ticket_upgrade_prediction/models/neural_net_torch/neural_net_model.py]: good point

File: ticket_upgrade_prediction/models/base.py
Comment (kuba1302) [reply ticket_upgrade_prediction/models/base.py:24]: It is used for typing purposed, in `neural_net_model.py`

File: ticket_upgrade_prediction/models/neural_net_torch/train.py
Comment (kuba1302) [reply ticket_upgrade_prediction/models/neural_net_torch/train.py]: good point

File: ticket_upgrade_prediction/models/neural_net_torch/train.py
Comment (kuba1302) [reply ticket_upgrade_prediction/models/neural_net_torch/train.py]: It is not, it should be `Optional[str]`

File: ticket_upgrade_prediction/models/neural_net_torch/train.py
Comment (kuba1302) [reply ticket_upgrade_prediction/models/neural_net_torch/train.py]: good point

File: ticket_upgrade_prediction/models/neural_net_torch/train.py
Comment (kuba1302) [reply ticket_upgrade_prediction/models/neural_net_torch/train.py:172]: Do we really need non stratified kfold?

File: ticket_upgrade_prediction/models/neural_net_torch/neural_net_model.py
Comment (jzmujdzin) [inline ticket_upgrade_prediction/models/neural_net_torch/neural_net_model.py]: i believe typing would be useful in this function, i.e. threshold: float = 0.5

**Review by jzmujdzin — COMMENTED** [review]
add typing pls to ticket_upgrade_prediction/models/neural_net_torch/train.py; fix conflicts

**Pull request merged**
